docs/states.md

```diff
@@ -0,0 +1,245 @@
+| `data-[…]:`          | (unsupported) `&[data-…]`                                      |
+| `open:`              | `&[open]`                                                      |
+
+
+## Pseudo-classes
+### Hover, focus, and active
+Style elements on hover, focus, and active using the `hover`, `focus`, and `active` modifiers.
+
+```html
+<div class="s-bg-positive hover:s-bg-positive-hover">
+...
+</div>
+```
+
+Warp also includes modifiers for other interactive states like `:visited`, `:focus-within`, `:focus-visible`, and more.
+
+### First, last, odd, and even
+Style an element when it is the first-child or last-child using the `first` and `last` modifiers.
+
+```html
+<ul role="list" class="p-8 ...">
+    <!-- Remove top/bottom padding when first/last child -->
+    <li class="flex py-8 first:pt-0 last:pb-0 ...">
+      ...
+    </li>
+    ...
+</ul>
+```
+
+You can also style an element when it’s an odd or even child using the `odd` and `even` modifiers
+
+```html
+<table>
+  <!-- ... -->
+  <tbody>
+      <!-- Use a white background for odd rows, and subtle for even rows -->
+      <tr class="odd:bg-white even:bg-subtle">
+        <td>{person.name}</td>
+        <td>{person.title}</td>
+        <td>{person.email}</td>
+      </tr>
+      ...
+  </tbody>
+</table>
+```
+
+Warp also includes modifiers for other structural pseudo-classes like `:only-child`, `:first-of-type`, `:empty`, and more.
+
+### Form states
+Style form elements in different states using modifiers like `required`, `invalid`, and `disabled`.
+
+```html
+<input
+  type="text"
+  value="stellar"
+  disabled
+  class="
+    w-full
+    bg-white
+    border
+    s-border
+    focus:s-border-focus
+    disabled:s-border-disabled
+    invalid:s-border-invalid
+"/>
+```
+
+Warp also includes modifiers for other form states like `:read-only`, `:indeterminate`, `:checked`, and more.
+
+We should always use the warp components for form elements if possible, they handle state styling internally by warp automagic.
+
+
+## Pseudo-elements
+### Before and after
+Style the ::before and ::after pseudo-elements using the `before` and `after` modifiers.
+
+```html
+  <span class="after:ml-16 ..."></span>
+```
+
+It’s worth noting that you don’t really need `::before` and `::after` pseudo-elements for most things in Warp projects — it’s usually simpler to just use a real HTML element.
+
+### Placeholder text
+Style the placeholder text of any input or textarea using the `placeholder` modifier.
+
+html```
+  <input class="placeholder:s-text-placeholder ... " placeholder="Search for anything..." type="text" name="dingbat"/>
+```
+
+### File input buttons
+Style the button in file inputs using the file modifier.
+
+html```
+<input type="file" class="file:mr-16 ... "/>
+```
+
+### List markers
+Style the counters or bullets in lists using the marker modifier.
+
+```html
+<ul role="list" class="marker:s-marker">
```

Comment:
Should we add this class to @warp-ds/drive (and token to @warp-ds/css/tokens)? We don't have support for `s-marker` at the moment.

Reply:
In dialogue with Adi about this one. I think ill just use s-text-current for now